app/Core/Router.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace App\Core;
+
+
+class Router
+{
+    protected $routes = [];
+    protected $params = [];
+    protected $request;
+
+    function __construct($request)
+    {
+        $routes = require APP_ROOT.'/routes/web.php';
+        foreach($routes as $route => $params) {
+            $this->routes["#^$route$#"] = $params;
+            $this->request = $request;
+        }
+    }
+
+    function match()
+    {
+        $url = $_REQUEST['route'];
+        foreach($this->routes as $route => $params) {
+            if (preg_match($route, $url, $matches)) {
+                $this->params = $params;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function run()
```

Comment:
This method can be splitted into multiple. 
Basically, every time you have more than 2 indentation levels it's a sign to extract a new method.

For now, it's just a recommendation, no actions are required.